### PR TITLE
AwsContext has a single setter for Credentials

### PR DESCRIPTION
**Pull request by awschristou**
## Description

This change adjusts AwsContext so that all Credential information is set atomically. Having multiple setters for various attributes is a maintenance risk for keeping them all in sync.

Out of Scope:
* AwsContext is currently used to retrieve any credentials. In a future change, this will be changed to only retrieve the active credential. Arbitrary credential loading/retrieval will be implemented in a separate change.

## Motivation and Context

In preparation for some new credentials handling code, some of the existing credentials related code flows are being tidied up in separate, isolated changes.

## Testing

There should be no change in functionality
* Signing into the toolkit updates the AWS Explorer and status bar appropriately
* Signing out of the toolkit updates the AWS Explorer and status bar appropriately
* Activating the toolkit (with previous credentials and without previous credentials) updates the AWS Explorer and status bar appropriately
* deploying a SAM Application and invoking a remote Lambda work appropriately

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/awsContext.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 import * as vscode from 'vscode'
 
+export interface AwsContextCredentials {
+    readonly credentials: AWS.Credentials
+    readonly credentialsId: string
```
Comment (bryceitoc9) [inline src/shared/awsContext.ts]: I'd recommend leaving this as profileName, it's still referred to as such everywhere else in the codebase.

File: src/test/shared/defaultAwsContext.test.ts [hunk 2]
```diff
@@ -159,31 +208,27 @@ describe('DefaultAwsContext', () => {
         assert.strictEqual(persistedRegions![0], testRegion1Value)
     })
 
-    it('fires event on profile change', async () => {
+    it('fires event on credentials change', async () => {
```
Comment (bryceitoc9) [inline src/test/shared/defaultAwsContext.test.ts]: Does this test pass if the event doesn't fire? If so, can you force a failure after the Promise if resolve() isn't called?

File: src/shared/awsContext.ts (hunk 1 above)
Comment (awschristou) [reply src/shared/awsContext.ts]: Over the course of bringing on a new credentials manager, we should transform code away from the concept of "profile" over to "credentials". Profile is a subset of Credentials, its just that its the only portion we've been working with so far. (The term 'profile' will likely be used with any credentials coming from the credentials/config shared ini files as part of the credentialsId, much like the JetBrains toolkit does)

File: src/test/shared/defaultAwsContext.test.ts (hunk 2 above)
Comment (awschristou) [reply src/test/shared/defaultAwsContext.test.ts]: If the event doesn't fire, the test fails with a timeout after two seconds, with a message along the lines of 

```
Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

**Pull request merged**